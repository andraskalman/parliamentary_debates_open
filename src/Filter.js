import React from 'react';
import { Radio } from 'react-bootstrap';
import './filter.css'

const Filter = ({ onChange, counts, filter_value, field_name, getLabel = (item => item.key), getValue = (item => item.key) }) => (
  <ul className="filter">
    <li key="">
        <Radio onChange={event => onChange({[field_name]: '',})} checked={filter_value === ''}>
          Ã–sszes
        </Radio>
    </li>
    {counts.map(item => (
      <li key={item.key}>
        <Radio onChange={event => onChange({ [field_name]: getValue(item), }
        )} checked={getValue(item) === filter_value}>
          {getLabel(item)} <b>({item.doc_count})</b>
        </Radio>
      </li>
    ))}
  </ul>
)

export default Filter
