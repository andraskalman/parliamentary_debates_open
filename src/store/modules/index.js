import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'
import search from './search'
import help from './help'

export default combineReducers({
  routing: routerReducer,
  search,
  help
})
