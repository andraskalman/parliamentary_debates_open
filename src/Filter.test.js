import React from 'react';
import ReactDOM from 'react-dom';
import Filter from './Filter';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import store, { history } from './store'


it('renders correctly', () => {
  const tree = renderer
    .create(<Filter
        counts={[
          {'key': 'John Doe', 'doc_count': '3'},
          {'key': 'asdfg', 'doc_count': '1'},
          {'key': 'sdgsdf', 'doc_count': '1'},
        ]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
