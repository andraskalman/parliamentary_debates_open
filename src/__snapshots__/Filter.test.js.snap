// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="filter"
>
  <li>
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="form-check-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
      />
      <label
        className="form-check-label"
        title=""
        type="checkbox"
      >
        Ã–sszes
      </label>
    </div>
  </li>
  <li>
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="form-check-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
      />
      <label
        className="form-check-label"
        title=""
        type="checkbox"
      >
        John Doe
         
        <b>
          (
          3
          )
        </b>
      </label>
    </div>
  </li>
  <li>
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="form-check-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
      />
      <label
        className="form-check-label"
        title=""
        type="checkbox"
      >
        asdfg
         
        <b>
          (
          1
          )
        </b>
      </label>
    </div>
  </li>
  <li>
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="form-check-input"
        disabled={false}
        onChange={[Function]}
        type="radio"
      />
      <label
        className="form-check-label"
        title=""
        type="checkbox"
      >
        sdgsdf
         
        <b>
          (
          1
          )
        </b>
      </label>
    </div>
  </li>
</ul>
`;
